src/server/api/procedures.ts

```diff
@@ -2,13 +2,15 @@ import { contextStorage } from '@/server/api/context';
 import { trpc } from '@/server/api/trpc';
 import { TRPCError } from '@trpc/server';
 
-const contextMiddleware = trpc.middleware((opts) => {
+// Makes the context available in input validation
+const procedureWithContext = trpc.procedure.use((opts) => {
   return contextStorage.run(opts.ctx, async () => {
     return await opts.next();
   });
 });
 
-const timingMiddleware = trpc.middleware(async ({ next, path }) => {
+// Use this procedure for any public endpoint. The timing here will add a varying delay so we will see the loading state when accessing the API in development.
+const publicProcedure = procedureWithContext.use(async ({ next, path }) => {
```

Comment:
Re-write all of these comments to JSDoc, so that they are available in IDEs as tooltips:
```suggestion
/**
 * Use this procedure for any public endpoint. The timing here will add a varying delay so we will see the loading state when accessing the API in development.
 */
const publicProcedure = procedureWithContext.use(async ({ next, path }) => {
```